docs/source/links/state.md

```diff
@@ -0,0 +1,295 @@
+    networkStatus(isConnected: $isConnected) @client {
+      isConnected
+    }
+  }
+`;
+```
+
+To fire off the mutation from your component, bind your mutation to your
+component via your favorite Apollo view layer integration just like you normally
+would. Here's what this would look like for React:
+
+```js
+const WrappedComponent = graphql(UPDATE_NETWORK_STATUS, {
+  props: ({ mutate }) => ({
+    updateNetworkStatus: isConnected => mutate({ variables: { isConnected } }),
+  }),
+})(NetworkStatus);
+```
+
+Now that you've seen how easy it is to update local state with
+`apollo-link-state`, let's dive deeper into the resolver map.
+
+<h2 id="resolver">Resolver map</h2>
+
+Your resolver map is where all the magic happens to retrieve and update your
+local data in the Apollo cache. The resolver map is an object with resolver
+functions for each GraphQL object type. You can think of a GraphQL query or
+mutation as a tree of function calls for each field.
+
+Each field must either resolve to a GraphQL primitive (Int, String, etc) or
+resolve in a function call. You don't have to specify resolver functions for
+every field, however. GraphQL has [default
+resolvers](docs/graphql-tools/resolvers.html#Default-resolver). If the return
+value from the parent object has the same property names as the fields requested
+in the child object, you won't need to specify a resolver.
+
+```js
+const getUser = gql`
+  query {
+    user(id: 1) {
+      name {
+        last
+        first
+      }
+    }
+  }
+`;
+```
+
+For this query, you will need to specify a resolver for `Query.user` in your
+resolver map. If `Query.user` returns an object with a name property that
+corresponds to an object with last and first properties, you do not need to
+specify any additional resolvers. GraphQL takes care of that for you!
+
+The three most important things to keep in mind about resolvers in
+`apollo-link-state` are this:
+
+1. The cache is added to the context for you so you can write & read data from
+   the cache.
+2. The resolver should return an object with a `__typename` property. This is
+   necessary for Apollo Client to [normalize the data in the
+   cache](/docs/react/basics/caching.html#normalization) properly.
+3. Resolver functions can be asynchronous if you need to perform side effects.
+
+If any of that sounds confusing, I promise it will be cleared up by the end of
+this section. Keep on reading! 😀
+
+<h3 id="async">Resolver signature</h3>
+
+The signature of a resolver function is the exact same as resolver functions on
+the server built with `graphql-tools`. Let's quickly recap the four parameters
+of a resolver function:
+
+```js
+fieldName(obj, args, context, info) => result
+```
+
+1. `obj`: The object containing the result returned from the resolver on the
+   parent field or the `ROOT_QUERY` object in the case of a top-level query or
+   mutation. Don't worry about this one - you probably won't need to use it for
```

Comment:
You definitely need this argument, you just don't need to worry about the root thing.

Reply:
Yeah I definitely could have phrased this better.